src/Translator.php

```diff
@@ -0,0 +1,542 @@
+                    "base" => "$currentBranch"
+                ], JSON_PRETTY_PRINT + JSON_UNESCAPED_SLASHES + JSON_UNESCAPED_UNICODE)
+            ]);
+            $code = $response->getStatusCode();
+            $responseBody = (string) $response->getBody();
+            if ($code === 201) {
+                $pullRequestUrl = json_decode($responseBody)->html_url;
+                $this->log("Pull request was succesfully created at $pullRequestUrl");
+                $this->pullRequestUrls[] = $pullRequestUrl;
+            } else {
+                $this->log($responseBody);
+                throw new Exception("Pull request failed - status code was $code");
+            }
+        }
+    }
+
+    private function outputPullRequestUrls(): void
+    {
+        $this->log('');
+        $this->log('The following pull-requests were created:');
+        foreach ($this->pullRequestUrls as $pullRequestUrl) {
+            $this->log($pullRequestUrl);
+        }
+        $this->log('');
+    }
+
+    private function getModuleName(string $modulePath): string
+    {
+        return json_decode(file_get_contents("$modulePath/composer.json"))->name;
+    }
+
+    private function exec(string $command, ?string $cwd = null): string
+    {
+        $this->log("Running $command");
+        $process = Process::fromShellCommandline($command, $cwd);
+        $process->run();
+        if (!$process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+        return trim($process->getOutput());
+    }
+
+    private function log(string $message): void
+    {
+        echo "$message\n";
+    }
+
+    /**
+     * Gets the module lang dir
+     */
+    private function getLangDirectory(string $modulePath): string
+    {
+        $sources = $this->getTransifexSources($modulePath);
+        foreach ($sources as $source) {
+            if (preg_match('#^(?<dir>.+)\\/(?<file>[^\\/]+)\\.yml$#', $source, $matches)) {
+                return $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Gets the directories of the JS lang folder.
+     */
+    private function getJSLangDirectories(string $modulePath): array
+    {
+        $sources = $this->getTransifexSources($modulePath);
+        $dirs = [];
+        foreach ($sources as $source) {
+            // Strip out /src/ dir and trailing file.js
+            if (preg_match('#^(?<dir>.+)\\/src\\/(?<file>[^\\/]+)\\.js(on)?$#', $source, $matches)) {
+                $dirs[] = $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return $dirs;
+    }
+
+    /**
+     * Get list of transifex source files. E.g. lang/en.yml
+     */
+    private function getTransifexSources(string $modulePath): array
+    {
+        $path = "$modulePath/.tx/config";
+        $content = file_get_contents($path);
+        $sources = [];
+        foreach (preg_split('~\R~u', $content) as $line) {
```

Comment:
```suggestion
        foreach (preg_split('#\R#u', $content) as $line) {
```
May as well standardise it to use the same style as all the other regex in this file, or else it looks like there's a _reason_ to use something different.

Reply:
Updated